app/views/ShareView/index.js

```diff
@@ -0,0 +1,93 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {
+	View, Text, TouchableOpacity, TextInput
+} from 'react-native';
+import ShareExtension from 'react-native-share-extension';
+import { HeaderBackButton } from 'react-navigation';
+
+import {
+	COLOR_TEXT_DESCRIPTION, HEADER_BACK
+} from '../../constants/colors';
+import I18n from '../../i18n';
+import RocketChat from '../../lib/rocketchat';
+import styles from './styles';
+
+export default class ShareView extends React.Component {
+	static navigationOptions = ({ navigation }) => ({
+		headerLeft: (
+			<HeaderBackButton
+				title={I18n.t('Back')}
+				backTitleVisible
+				onPress={navigation.goBack}
+				tintColor={HEADER_BACK}
+			/>
+		),
+		title: I18n.t('Share'),
+		headerRight: (
+			<TouchableOpacity onPress={navigation.getParam('sendMessage')} style={styles.sendButton}>
+				<Text style={styles.send}>{I18n.t('Send')}</Text>
+			</TouchableOpacity>
+		)
+	})
+
+	static propTypes = {
+		navigation: PropTypes.object
+	};
+
+	constructor(props) {
+		super(props);
+		this.state = {
+			rid: '',
+			text: '',
+			name: ''
+		};
+	}
+
+	componentWillMount() {
+		const { navigation } = this.props;
```

Comment:
can you move this entire logic to constructor?

Reply:
Set State is not possible on constructor :/

Reply:
`const rid = navigation.getParam('rid', '');
 const name = navigation.getParam('name', '');
 const text = navigation.getParam('text', '');
 this.state = {
     rid,
     name,
     text
}

Reply:
why not this

Reply:
oh! nice, sorry! haha <3